rust/src/main.rs

```diff
@@ -29,13 +29,27 @@ use std::path::{Path, PathBuf};
 use xi_core_lib::XiCore;
 use xi_rpc::RpcLoop;
 
-const XI_DIRECTORY: &str = "xi-core";
+const XI_LOG_DIR: &str = "xi-core";
 const XI_LOG_FILE: &str = "xi-core.log";
 
-fn get_logging_directory() -> Result<PathBuf, io::Error> {
+fn get_logfile_directory_name() -> String {
+    match std::env::var("XI_LOG_DIR") {
+        Ok(name) => name,
+        Err(_) => String::from(XI_LOG_DIR),
+    }
+}
+
+fn get_logfile_file_name() -> String {
```

Comment:
both this and the previous fn could return `PathBuf` instead of `String`. These fns also mean we're always allocating, even when we're just using the statics, but we probably don't need to sweat a few tens of ns here just yet, and this is nice and clear.